runtime/src/iree/hal/drivers/hip/hip_allocator.c

```diff
@@ -114,6 +158,27 @@ static void iree_hal_hip_allocator_destroy(
       iree_hal_hip_allocator_cast(base_allocator);
   IREE_TRACE_ZONE_BEGIN(z0);
 
+  iree_slim_mutex_deinitialize(&allocator->async_allocation_mutex);
```

Comment:
fyi it's generally a good idea to only release a mutex that guards a resource after you clean up the resource (otherwise it's possible functions/callbacks that are called as cleanup may assume it's still valid) - it can also appease tsan (which I wish we ran) as it often doesn't like seeing locked access on thread 2 and then unlocked access on thread 1 (it doesn't know you're deinitializing, only that you aren't using a lock here)